### PR TITLE
bayesian optimization draft

### PR DESCRIPTION
**Pull request by libbyandhelen**
This is a draft for bayesian optimization (work in progress) #15 @gaocegege @YujiOshima

### CODE REVIEW

**Comment by gaocegege** [comment]
Thanks for your awesome contribution! I will take a look soon.

**Comment by gaocegege** [comment]
When the PR is ready to merge, please use `/hold cancel` to cancel the hold.

File: suggestion/python/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+- start the service
+
+```
+python suggestion/bayesian/main.py
+```
+
+- start the testing client
+
+
```
Comment (gaocegege) [inline suggestion/python/README.md]: Please remove the new line here.

File: suggestion/python/api/api.proto [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+	rpc CompleteTrial(CompleteTrialRequest) returns (CompleteTrialReply);
+	rpc ShouldTrialStop(ShouldTrialStopRequest) returns (ShouldTrialStopReply);
+	rpc GetObjectValue(GetObjectValueRequest) returns (GetObjectValueReply);
+	rpc AddMeasurementToTrials(AddMeasurementToTrialsRequest) returns (AddMeasurementToTrialsReply);
+    rpc InitializeSuggestService(InitializeSuggestServiceRequest) returns(InitializeSuggestServiceReply);
+}
+
+service Suggestion {
+	rpc GenerateTrials(GenerateTrialsRequest) returns (GenerateTrialsReply);
+    rpc SetSuggestionParameters(SetSuggestionParametersRequest) returns (SetSuggestionParametersReply);
```
Comment (gaocegege) [inline suggestion/python/api/api.proto]: Inconsistent blanks here. Please fix it.

**Comment by gaocegege** [comment]
/ok-to-test

Please rebase the master and I think the test should pass.

**Comment by YujiOshima** [comment]
@libbyandhelen Please move `suggestion/python/api/` directory to `api/python` and store your bayesian-optimization source codes under `suggestion/bayesianoptimization` directory in conformity to other services.

**Comment by YujiOshima** [comment]
@libbyandhelen And please dockerize your suggestion service and add k8s manifest.
I'm sorry there is no nice documents. Please refer other services.
https://github.com/kubeflow/hp-tuning/blob/master/suggestion/random/Dockerfile
https://github.com/kubeflow/hp-tuning/tree/master/manifests/vizier/suggestion/random

**Comment by gaocegege** [comment]
@libbyandhelen If you have any question about `dockerize your suggestion service and add k8s manifest` please at me and I can help you.

**Comment by libbyandhelen** [comment]
@gaocegege I wrote both the dockerfile and k8s manifest. I tested the dockerfile and it works fine, but I do not know exactly how to test the k8s manifest. Could you take a look of my code to see whether it is OK?

**Comment by libbyandhelen** [comment]
@gaocegege OK, got it.
@YujiOshima 
Do you think the api proto is OK now? I did not change the protocol in api.proto.
I used `SetSuggestionParameters` to set all the parameters needed for the suggestion service, and `StopSuggestion` for deleting the relevant information of the specific study stored in the service.
`completed_trials` in `GenerateTrials` is used to pass information of trials for now, since the interface for DB is not ready.

**Comment by YujiOshima** [comment]
@libbyandhelen OK, we will add DB interface for python later.
Finally, please add build and deploy test for this suggestion to test script.
Since this is in python, you don't need `go build` test. Only `gcloud container builds submit` and `deploy`.
https://github.com/kubeflow/hp-tuning/blob/master/test/scripts/build.sh
https://github.com/kubeflow/hp-tuning/blob/master/test/scripts/run-tests.sh

**Comment by gaocegege** [comment]
I think it is the last step! 🎉 

Thanks for your contribution!

**Comment by YujiOshima** [comment]
@libbyandhelen Thank you for your great contribution!
/lgtm

**Pull request merged**